### PR TITLE
add scalafix+scalameta+scalaparse+scalatex+ammonite-ops+geny+scalaj-http

### PR DESCRIPTION
**Pull request by SethTisue**
this rabbit hole was deep.

### CODE REVIEW

**Comment by SethTisue** [comment]
tested locally:

```text
[info] Project acyclic-----------------: SUCCESS (unchanged, not rebuilt)
[info] Project akka--------------------: SUCCESS (unchanged, not rebuilt)
[info] Project fastparse---------------: SUCCESS (unchanged, not rebuilt)
[info] Project genjavadoc--------------: SUCCESS (unchanged, not rebuilt)
[info] Project geny--------------------: SUCCESS (unchanged, not rebuilt)
[info] Project macro-paradise----------: SUCCESS (unchanged, not rebuilt)
[info] Project scala-------------------: SUCCESS (unchanged, not rebuilt)
[info] Project scala-java8-compat------: SUCCESS (unchanged, not rebuilt)
[info] Project scala-js----------------: SUCCESS (unchanged, not rebuilt)
[info] Project scala-parser-combinators: SUCCESS (unchanged, not rebuilt)
[info] Project scalacheck--------------: SUCCESS (unchanged, not rebuilt)
[info] Project scalafix----------------: SUCCESS (project rebuilt ok)
[info] Project scalameta---------------: SUCCESS (unchanged, not rebuilt)
[info] Project scalatest---------------: SUCCESS (unchanged, not rebuilt)
[info] Project sourcecode--------------: SUCCESS (unchanged, not rebuilt)
[info] Project utest-------------------: SUCCESS (unchanged, not rebuilt)
[info] >>> The dbuild result is------------: SUCCESS (project rebuilt ok)
```

**Comment by SethTisue** [comment]
fyi @lihaoyi @olafurpg @xeno-by @hoffrocket

File: community.dbuild [hunk 1]
```diff
@@ -984,4 +989,60 @@ build += {
     extra.exclude: ["plugin"]  // we never build sbt plugins
   }
 
+  // forked to remove the usual bintray gunk
```
Comment (xeno-by) [inline community.dbuild:992]: Can we provide some additional configuration on our side (e.g. `-Dsbt.prohibit.publish=true`) to make sure you don't need to maintain your fork?

Comment (SethTisue) [reply community.dbuild:992]: with this particular plugin, I don't think so, sadly. a bunch of other projects are affected, too. the ticket is https://github.com/typesafehub/dbuild/issues/158

Comment (SethTisue) [reply community.dbuild:992]: (I actually did try `-Dsbt.prohibit.publish=true` before forking, to no avail.)

File: community.dbuild [hunk 2]
```diff
@@ -984,4 +989,60 @@ build += {
...
+  // forked to remove wartremover and coursier
+  ${vars.base} {
+    name: "scalafix"
+    uri:  ${vars.uris.scalafix-uri}
+    extra.exclude: [
+      "scalafix-sbt"  // we never build sbt plugins
+      "scalafix-nsc"  // depends on circe which we don't have yet
+      // this didn't work; I decided to postpone any investigation. dbuild output an
+      // immensely long error (tens of megabytes), enough to choke Jenkins
+      "scalafix-tests"
```
Comment (olafurpg) [inline community.dbuild:1010]: Yes, I would exclude this directory. Here's where I run rewrites against a mini scalafix community-build. This involves cloning other projects and compiling them.

File: community.dbuild [hunk 3]
```diff
@@ -984,4 +989,60 @@ build += {
...
+    extra.exclude: ["readme"]  // requires scalatex-site which requires Scala.js
+  }
+
+  // forked to remove wartremover and coursier
+  ${vars.base} {
+    name: "scalafix"
+    uri:  ${vars.uris.scalafix-uri}
+    extra.exclude: [
+      "scalafix-sbt"  // we never build sbt plugins
+      "scalafix-nsc"  // depends on circe which we don't have yet
```
Comment (olafurpg) [inline community.dbuild:1007]: Note, most scalafix tests are currently in scalafix-nsc, core has a limited amount of tests. These tests depend on various projects (circe, shapeless,...) to reproduce bugs that surfaced in larger integration tests.

**Review by olafurpg — COMMENTED** [review]
Thanks for the heads up! Just some FYIs

**Comment by SethTisue** [comment]
thanks Ólafur, I added your comments in 56c9c953512510d1ea0952b052e34f385b5d1886

**Pull request merged**